scalafmt-core/shared/src/main/scala/org/scalafmt/config/RewriteSettings.scala

```diff
@@ -21,4 +22,12 @@ case class RewriteSettings(
       )
   }
 
+  if (sortModifiers.order.distinct.length != 8)
```

Comment:
It's better to write a decoder by hand in the `SortSettings` companion, the values are encoded from `Conf.Lst(values)` where values should be only `Conf.Str`.

The `Rewrite.validateRewrites(rules)` validation above is a hack that should not be repeated, I'm cleaning up configuration in #1145.

Reply:
Actually, this is fine, I can take care of it in #1145 once this PR is merged :)